src/PhpFlo/Common/DefinitionInterface.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/*
+ * This file is part of the <package> package.
+ *
+ * (c) Marc Aschmann <maschmann@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+declare(strict_types=1);
+namespace PhpFlo\Common;
+
+/**
+ * Interface DefinitionInterface
+ *
+ * @package PhpFlo\Common
+ * @author Marc Aschmann <maschmann@gmail.com>
+ */
+interface DefinitionInterface
+{
+    /**
+     * @param array $definition
+     * @return DefinitionInterface
+     */
+    public function definition(array $definition) : DefinitionInterface;
+
+    /**
+     * @return array
+     */
+    public function properties() : array;
+
+    /**
+     * @return array
+     */
+    public function initializers() : array;
+
+    /**
+     * @return array
+     */
+    public function processes() : array;
+
+    /**
+     * @return array
+     */
+    public function connections() : array;
+
+    /**
+     * @return array
+     */
+    public function toArray() : array;
+
+    /**
+     * @return string
+     */
+    public function toJson() : string;
```

Comment:
nothing official,yet - but it seems to condense around ():<space><type>
Adjusted accordingly :-)